### DISCUSSION TITLE
Feature request: Import and Export palettized bmp and png formats

### DISCUSSION THREAD

**Discussion by mrgudenheim** [Orama-Interactive/Pixelorama#1130] [Ideas]
Feature request to allow importing and exporting .bmp and .png files with 4-bit or 8-bit bit palletized format that includes palette data for up to 256 colors. When importing there would be a way to import the entire palette including colors that are not used in the image or to selectively overwrite the existing palette with colors from the imported file (similar to how GraphicsGale does it). When exporting, a user could choose the 8-bit per pixel (8bpp) or 4-bit (4bpp) bmp or png format to include the current palette in the file data.

**Comment by alope107** [comment]
I'd like to echo this request for 8bpp bmp export. I use Butano for GameBoy Advance development and it requires this file format. As it stands I need to do a bit of annoying postprocessing to convert the exported pngs to bmps, and this is preventing me from including Pixelorama in a bundled set of tools for GBA development.

I'd be happy to take a stab at implementing this if there is support from the maintainers for it.

**Reply by OverloadedOrama** [reply]
> I'd be happy to take a stab at implementing this if there is support from the maintainers for it.

Sure thing! Feel free to implement it :)

**Comment by thirsty-wing** [comment]
I echo this as well. My use case is when using this with SGDK https://github.com/Stephane-D/SGDK, which requires sprites and backgrounds to be palettized pngs. I would expect a palettized project to be exported as a palettized png by default. Or, that there should be a way to opt in to exporting as a palettized png. Thank you!
